tgui/packages/tgui/interfaces/ModularShieldGen.tsx

```diff
@@ -0,0 +1,142 @@
+import { useBackend } from '../backend';
+import { Window } from '../layouts';
+import { Stack, Section, ProgressBar, Button, NumberInput, LabeledList } from '../components';
+import { BooleanLike } from 'common/react';
+
+type ModularShieldGenData = {
+  max_strength: number;
+  current_strength: number;
+  max_regeneration: number;
+  current_regeneration: number;
+  max_radius: number;
+  current_radius: number;
+  active: BooleanLike;
+  recovering: BooleanLike;
+  exterior_only: BooleanLike;
+  initiating_field: BooleanLike;
+};
+
+export const ModularShieldGen = (props, context) => {
+  const { topLevel } = props;
+  const { act, data } = useBackend<ModularShieldGenData>(context);
+  const {
+    max_strength,
+    max_regeneration,
+    current_regeneration,
+    max_radius,
+    current_radius,
+    current_strength,
+    active,
+    exterior_only,
+    recovering,
+    initiating_field,
+  } = data;
+
+  return (
+    <Window title="Modular Shield Generator" width={600} height={400}>
+      <Window.Content scrollable>
```

Comment:
```suggestion
      <Window.Content>
          <Stack fill>
           <Stack.Item grow={2}>
```

experiment with different grow percentages to get the desired result